### PR TITLE
finished resolvers for create update softdelete request

### PR DESCRIPTION
**Pull request by Anand-Issac**
This PR creates resolvers for requests to be able to create, update, and soft delete.

Creating request in playground + verifying they are created
Create
Created

Updating request in playground + verifying they are updated
Update
Updated

Deleting request in playground + verifying they are deleted
Delete
Deleted

### CODE REVIEW

File: server/graphql/schema.ts [hunk 1]
```diff
@@ -53,6 +58,23 @@ const typeDefs = gql`
...
+        softDeleteRequest(id: ID): ServerResponse
+    }
+    input ClientInput {
+        _id: ID
+        clientId: String
+        firstName: String
+        lastName: String    
+    }
+    input RequestInput {
+        client: ClientInput
```
Comment (guy-ro) [inline server/graphql/schema.ts]: You should be able to create a Request without creating a new Client. Please change the type of client to ID.

File: server/datasources/requestDataSource.ts [hunk 2]
```diff
@@ -1,9 +1,73 @@
...
 import { RequestDocument } from '../models/requestModel'
+import { ServerResponseInterface } from '../graphql/serverResponse'
+
+import { Types } from 'mongoose'
 
 export default class RequestDataSource extends CachedMongooseDataSource<RequestDocument> {
   constructor() {
     super(RequestCache)
   }
+  async createRequest(request: RequestDocument): Promise<ServerResponseInterface> {
```
Comment (guy-ro) [inline server/datasources/requestDataSource.ts]: If we are putting these generic functions in cachedMongooseDataSource, please follow up with this and modify/remove these functions accordingly.

File: server/graphql/resolvers.ts [hunk 3]
```diff
@@ -15,6 +16,11 @@ const resolvers = {
     requestGroup: (_, { id }, { dataSources }): RequestGroupInterface => dataSources.requestGroups.getById(Types.ObjectId(id)),
     requestGroups: (_, __, { dataSources }): Array<RequestGroupInterface> => dataSources.requestGroups.getAll()
   },
+  Mutation: {
+    createRequest: (_, { request }, { dataSources }): Promise<ServerResponseInterface> => dataSources.requests.createRequest(request),
+    updateRequest: (_, { request, id }, { dataSources }): Promise<ServerResponseInterface> => dataSources.requests.updateRequest(request, id),
```
Comment (guy-ro) [inline server/graphql/resolvers.ts]: The id for the request to update should probably be supplied in the request object itself

**Review by guy-ro — CHANGES_REQUESTED** [review]
There are a few things that should be changed/discussed. Besides that, looks good!

**Review by meganniu — COMMENTED** [review]
I'll be reviewing this again after #61 is merged.
But looks great so far! @Anand-Issac

File: server/graphql/schema.ts (hunk 1 above)
Comment (Anand-Issac) [reply server/graphql/schema.ts]: Done

File: server/datasources/requestDataSource.ts (hunk 2 above)
Comment (Anand-Issac) [reply server/datasources/requestDataSource.ts]: Done

File: server/graphql/resolvers.ts (hunk 3 above)
Comment (Anand-Issac) [reply server/graphql/resolvers.ts]: Done

Comment (meganniu) [reply server/graphql/resolvers.ts]: LOL @Anand-Issac i don't know if the changes got pushed for this

Comment (Anand-Issac) [reply server/graphql/resolvers.ts]: @meganniu  oops

Comment (meganniu) [reply server/graphql/resolvers.ts]: nw :))

**Pull request closed without merging**